quickwit/quickwit-indexing/failpoints/mod.rs

```diff
@@ -334,14 +333,15 @@ async fn test_merge_executor_controlled_directory_kill_switch() -> anyhow::Resul
         after_universe_kill_clone.wait();
     })
     .unwrap();
+    fail::cfg(
+        "after-merge-split",
+        "panic(merge should be failed by directory kill switch)",
+    )
+    .unwrap();
     merge_executor_mailbox.send_message(merge_scratch).await?;
     before_universe_kill.wait();
     universe.kill();
     after_universe_kill.wait();
-    fail::cfg("before-merge-split", "off").unwrap();
-
-    let (exit_status, _) = merge_executor_handle.join().await;
-    assert!(matches!(exit_status, ActorExitStatus::Failure(_)));
```

Comment:
@fulmicoton-dd I know you wrote this a long time ago, but do you happen to remember why you were testing the `merge_executor_handle.join()` result instead of using a panic failpoint?

Note: the actor exit status is now `Killed` instead of `Failure` because since https://github.com/quickwit-oss/quickwit/pull/5704, merge failures return `Ok()` with an `error!`. log instead of failing the actor. But this is not what this test is about, hence the proposed change.